paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -260,125 +259,137 @@ def paasta_mark_for_deployment(args):
             component='deploy',
-            line=line,
+            line=("Couldn't find any instances for service {0} in deploy "
+                  "group {1}".format(service, deploy_group)),
             level='event'
         )
         raise NoInstancesFound
     paasta_print("Waiting for deployment of {0} for '{1}' complete..."
                  .format(git_sha, deploy_group))
     for cluster in cluster_map.values():
-        cluster['deployed'] = 0
-    try:
-        with Timeout(seconds=timeout):
-            total_instances = sum([len(v["instances"]) for v in cluster_map.values()])
-            with progressbar.ProgressBar(maxval=total_instances) as bar:
-                while True:
-                    for cluster, instances in cluster_map.items():
-                        if cluster_map[cluster]['deployed'] != len(cluster_map[cluster]['instances']):
-                            cluster_map[cluster]['deployed'] = instances_deployed(
-                                cluster=cluster,
-                                service=service,
-                                instances=instances['instances'],
-                                git_sha=git_sha)
-                            if cluster_map[cluster]['deployed'] == len(cluster_map[cluster]['instances']):
-                                instance_csv = ", ".join(cluster_map[cluster]['instances'])
-                                paasta_print("Deploy to %s complete! (instances: %s)" % (cluster, instance_csv))
-                        bar.update(sum([v["deployed"] for v in cluster_map.values()]))
-                    if all([cluster['deployed'] == len(cluster["instances"]) for cluster in cluster_map.values()]):
-                        sys.stdout.flush()
-                        break
-                    else:
-                        time.sleep(10)
-                    sys.stdout.flush()
-    except TimeoutError:
-        line = "\n\nTimed out after {0} seconds, waiting for {2} in {1} to be deployed by PaaSTA. \n\n"\
-               "This probably means the deploy hasn't suceeded. The new service might not be healthy or one "\
-               "or more clusters could be having issues.\n\n"\
-               "To debug: try running:\n\n"\
-               "    paasta status -s {2} -vv\n"\
-               "    paasta logs -s {2}\n\n"\
-               "to determine the cause.\n\n"\
-               "If the service is known to be slow to start you may wish to increase "\
-               "the timeout on this step.".format(timeout, deploy_group, service)
-        _log(
-            service=service,
-            component='deploy',
-            line=line,
-            level='event'
-        )
-        raise
-    return True
+        cluster['deployed'] = []
+    total_instances = sum([len(v["instances"]) for v in cluster_map.values()])
+    deadline = time.time() + timeout
+
+    with progressbar.ProgressBar(maxval=total_instances) as bar:
+        while time.time() < deadline:
```

Comment:
This progress bar doesn't take time into account at all.
https://github.com/Yelp/paasta/pull/944/files#diff-c18610688c26664370ccca67b0151768R368

It calculates percentage of deployed instances, and it can only be at 100% if all instances are deployed.